### DISCUSSION TITLE
Building Carla on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by erikgust2** [carla-simulator/carla#5509] [Q&A]
Hello!

I am currently attempting to build and run Carla on an installation of Ubuntu 22.04. However, it seems like getting Clang-8 to install on that release is exceedingly difficult, at least for someone who's relatively new to Linux. When running the "sudo apt-get install clang-8"-command in the command line, this is the result:

```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 clang-8 : Depends: libstdc++-5-dev but it is not installable
           Depends: libgcc-5-dev but it is not installable
           Depends: libobjc-5-dev but it is not installable
           Depends: libclang-common-8-dev (= 1:8.0.1+svn369350-1~exp1~20200112113617.82) but it is not going to be installed
           Recommends: libomp-8-dev but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

I can't seem to find anyone else running into this issue when googling it, so I'm assuming that I am missing something very obvious. If I understand the response correctly, Clang-8 itself seems to be available, but not it's dependencies, right?

Is there a way of getting around this or should I just give in and downgrade to 18.04? Sorry if it's a dumb question with an obvious answer. I am kinda learning Linux on the fly.

**Comment by fgoudreault** [comment]
clang-8 is not supported anymore on ubuntu 20 and above. Either you need a workaround to force install this compiler or build in docker.

**Comment by diff8911** [comment]
hello, I met the same question on Ubuntu20, according to [issuse 3441](https://github.com/carla-simulator/carla/issues/3441), I removed"deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-8 main" from /etc/apt/sources.list and running apt-get update i was able to install clang-8.
Hope this will help you.

**Reply by erikgust2** [reply]
hmm doesn't seem like that particular line exists in the Ubuntu 22.04 sources.list. I appreciate the tip though! :)

**Reply by ars-21** [reply]
Hi,
Were you able to build carla-simulator successfully on Ubuntu 22.04?
And is it fully functional?

It would be gr8 help if you could guide on the same
Thanks
Aryan Shah

**Reply by Piki1989** [reply]
Hi guys,
I'm also looking forward for such information.
I was able to compile UE4.26. Now I'm figuring how to build Carla with clang 11. Any ideas ?
Installing clang 8 on Ubuntu 22 is not possible :(

**Reply by srajangoyal** [reply]
Hi!
Has anybody found a solution for this issue?

**Reply by fgoudreault** [reply]
I compiled using docker therefore I could use ubuntu 18 directly and it worked.

**Reply by jumpymonkey1** [reply]
Do you happen to have a link or a guide you followed to install this?

**Reply by fgoudreault** [reply]
Unfortunately I don't have a guide for this as I scrambled a dockerfile myself in the end (which I don't have anymore). However, I managed to compile the unreal engine 4 using [this tool](https://docs.adamrehn.com/ue4-docker/read-these-first/introduction-to-ue4-docker). Then I built Carla on top of this image. Hope this helps

**Reply by fgoudreault** [reply]
You can also use the dockerfiles used by carla itself. Take a look [on them](https://github.com/carla-simulator/carla/tree/master/Util/Docker)!
